core/domain/blog_domain.py

```diff
@@ -0,0 +1,609 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain objects relating to blogs."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import json
+import re
+import string
+
+from constants import constants
+from core.domain import html_cleaner
+from core.domain import user_services
+
+import python_utils
+import utils
+
+
+class BlogPost(python_utils.OBJECT):
+    """Domain object for an Oppia Blog Post."""
+
+    def __init__(
+            self, blog_post_id, author_id, title, content, url_fragment, tags,
+            thumbnail_filename=None, last_updated=None, published_on=None):
+        """Constructs a BlogPost domain object.
+
+        Args:
+            blog_post_id: str. The unique ID of the blog post.
+            author_id: str. The user ID of the author.
+            title: str. The title of the blog post.
+            content: text. The html content of the blog post.
+            published_on: datetime.datetime. Date and time when the blog post is
+                last published.
+            last_updated: datetime.datetime. Date and time when the blog post
+                was last updated.
+            thumbnail_filename: str|None. The thumbnail filename of blog post .
+            url_fragment: str. The url fragment for the blog post.
+            tags: list(str). The list of tags for the blog post.
```

Comment:
follow the order of function agrs and place it simialrly

Reply:
Done.